### ISSUE TITLE
racket-8.1: `make cs RACKET=racket` fails on Fedora with racket-7.4: embed-boot.rkt: remove-signature: unbound identifier

### ISSUE THREAD

**Issue by juhp** [racket/racket#3911]
**What version of Racket are you using?**
Building racket-8.1 off racket-7.4

**What program did you run?**
`make cs RACKET=racket`

**What should have happened?**
build to succeed

**If you got an error message, please include it here.**
```
:
ChezScheme/ta6le/bin/ta6le/scheme -B ChezScheme/ta6le/boot/ta6le/petite.boot -B ChezScheme/ta6le/boot/ta6le/scheme.boot --script ./to-vfasl.ss   racket.boot racket-v.boot petite scheme
racket -O 'info@compiler/cm' -l- setup --boot ./../../setup-go.rkt ./compiled ignored ./ignored.d ./embed-boot.rkt --expect-elf --target ta6le  raw_racketcs racketcs petite-v.boot scheme-v.boot racket-v.boot
compiler/cm:   start-compile: /var/home/petersen/fedora/fork/danwi/racket/racket-8.1/src/cs/c/embed-boot.rkt
compiler/cm:   | start-compile: /usr/share/racket/collects/racket/cmdline.rkt
compiler/cm:   | finish-compile: /usr/share/racket/collects/racket/cmdline.rkt
compiler/cm:   | start-compile: /usr/share/racket/collects/compiler/private/pe-rsrc.rkt
compiler/cm:   | finish-compile: /usr/share/racket/collects/compiler/private/pe-rsrc.rkt
compiler/cm:   | start-compile: /usr/share/racket/collects/compiler/private/elf.rkt
compiler/cm:   | finish-compile: /usr/share/racket/collects/compiler/private/elf.rkt
compiler/cm:   | start-compile: /var/home/petersen/fedora/fork/danwi/racket/racket-8.1/src/cs/c/adjust-compress.rkt
compiler/cm:   |   start-compile: /usr/share/racket/collects/racket/system.rkt
compiler/cm:   |  |  start-compile: /usr/share/racket/collects/racket/private/streams.rkt
compiler/cm:   |  |  finish-compile: /usr/share/racket/collects/racket/private/streams.rkt
compiler/cm:   |   finish-compile: /usr/share/racket/collects/racket/system.rkt
compiler/cm:   |   start-compile: /usr/share/racket/collects/file/gzip.rkt
compiler/cm:   |   finish-compile: /usr/share/racket/collects/file/gzip.rkt
compiler/cm:   |   start-compile: /usr/share/racket/collects/file/gunzip.rkt
compiler/cm:   |   finish-compile: /usr/share/racket/collects/file/gunzip.rkt
compiler/cm:   |   start-compile: /usr/share/racket/collects/ffi/unsafe.rkt
compiler/cm:   |   finish-compile: /usr/share/racket/collects/ffi/unsafe.rkt
compiler/cm:   | finish-compile: /var/home/petersen/fedora/fork/danwi/racket/racket-8.1/src/cs/c/adjust-compress.rkt
compiler/cm:   finish-compile: /var/home/petersen/fedora/fork/danwi/racket/racket-8.1/src/cs/c/embed-boot.rkt
embed-boot.rkt:61:9: remove-signature: unbound identifier
  in: remove-signature
  location...:
   embed-boot.rkt:61:9
  context...:
   do-raise-syntax-error
   for-loop
   finish-bodys
   for-loop
   finish-bodys
   for-loop
   finish-bodys
   loop
   [repeats 2 more times]
   finish-expanding-body33
   lambda-clause-expander
   for-loop
   [repeats 1 more time]
   finish-bodys
   for-loop
   finish-bodys
   ...
make[3]: *** [Makefile:214: racketcs] Error 1
make[3]: Leaving directory '/var/home/petersen/fedora/fork/danwi/racket/racket-8.1/src/cs/c'
make[2]: *** [Makefile:88: cs] Error 2
make[2]: Leaving directory '/var/home/petersen/fedora/fork/danwi/racket/racket-8.1/src/cs/c'
make[1]: *** [Makefile:81: racketcs] Error 2
make[1]: Leaving directory '/var/home/petersen/fedora/fork/danwi/racket/racket-8.1/src'
make: *** [Makefile:85: cs] Error 2
```
**Please include any other relevant details**
Fedora Linux 33 and 34

See also https://bugzilla.redhat.com/show_bug.cgi?id=1863004 for more details.

**Comment by mflatt** [comment]
Using `RACKET=racket` only works when `racket` is the same version as the one being compiled. It's intended more for cross-compilation than boostrapping.

To use an existing, older Racket just for bootstrap, use `RACKET_FOR_BOOTFILES=racket`.

I see that our current v8.2 release candidate's bootstrap needs Racket v8.1.0.x, though. I'll work to get that regression fixed before the v8.2 release.

**Comment by juhp** [comment]
Thank you very much, that is very helpful - I am quite new to Racket!
